azurelinuxagent/common/cgroupapi.py

```diff
@@ -54,9 +57,12 @@ def cgroups_supported():
         distro_name = distro_info[0]
         try:
             distro_version = FlexibleVersion(distro_info[1])
+            major_minor_version = "{0}.{1}".format(distro_version.major, distro_version.minor)
         except ValueError:
             return False
-        return distro_name.lower() == 'ubuntu' and distro_version.major >= 16
+        return ((distro_name.lower() == 'ubuntu' and distro_version.major >= 16) or
+                (distro_name.lower() in ("centos", "redhat") and distro_version.major >= 7 and
+                 major_minor_version != "7.1"))
```

Comment:
this seems like it could get unwieldy... Can we move this into a function inside the OSUtil class heirarchy? that way we can keep this logic close to the distro itself.

Reply:
@kevinclark19a I like the idea, moving to each OSUtil and create abstract function quite bit of change, I will keep in mind next time.